src/scripts/main.js

```diff
@@ -1,3 +1,332 @@
+      if (probabilityCount < 10) {
+        board[rowInd][cellInd] = 2;
+        probabilityCount++;
+      } else {
+        board[rowInd][cellInd] = 4;
+        probabilityCount = 0;
+      }
+
+      return;
+    }
+  }
+};
+
+function shiftRowLeft(row) {
+  const newRow = [];
+
+  for (let j = 0; j < row.length; j++) {
+    if (row[j] !== 0) {
+      newRow.push(row[j]);
+    }
+  }
+
+  for (let j = newRow.length; j < row.length; j++) {
+    newRow.push(0);
+  }
+
+  return newRow;
+}
+
+function moveRight(check) {
+  for (let i = 0; i < board.length; i++) {
+    board[i] = shiftRowLeft(board[i]).reverse();
+  }
+
+  for (let i = 0; i < board.length; i++) {
+    for (let j = 1; j < board[i].length; j++) {
+      if (board[i][j - 1] === board[i][j]) {
+        board[i][j - 1] = 0;
+        board[i][j] = board[i][j] * 2;
+        score += board[i][j];
+      }
+    }
+  }
+
+  for (let i = 0; i < board.length; i++) {
+    board[i] = shiftRowLeft(board[i]).reverse();
+  }
+}
+
+function checkAndAct(check) {
+  if (!check) {
+    newCell();
+    display();
+    checkWin();
+  }
+}
+
+function moveLeft(check) {
+  for (let i = 0; i < board.length; i++) {
+    board[i] = shiftRowLeft(board[i]);
+  }
+
+  for (let i = 0; i < board.length; i++) {
+    for (let j = 1; j < board[i].length; j++) {
+      if (board[i][j - 1] === board[i][j]) {
+        board[i][j - 1] = board[i][j - 1] * 2;
+        board[i][j] = 0;
+        score += board[i][j - 1];
+      }
+    }
+  }
+
+  for (let i = 0; i < board.length; i++) {
+    board[i] = shiftRowLeft(board[i]);
+  }
+
+  if (!check) {
+    checkAndAct(check);
+  }
+}
+
+function moveUp(check) {
+  for (let i = 0; i < board.length; i++) {
+    let newColumn = [];
+
+    for (let j = 0; j < board[0].length; j++) {
+      newColumn.push(board[j][i]);
+    }
+
+    newColumn = shiftRowLeft(newColumn);
+
+    for (let j = 1; j < newColumn.length; j++) {
+      if (newColumn[j - 1] === newColumn[j]) {
+        newColumn[j - 1] = newColumn[j - 1] * 2;
+        newColumn[j] = 0;
+        score += newColumn[j - 1];
+      }
+    }
+
+    newColumn = shiftRowLeft(newColumn);
+
+    for (let j = 0; j < newColumn.length; j++) {
+      board[j][i] = newColumn[j];
+    }
+  }
+
+  if (!check) {
+    checkAndAct(check);
+  }
+}
+
+function moveDown(check) {
+  for (let i = 0; i < board.length; i++) {
+    let newColumn = [];
+
+    for (let j = 0; j < board[0].length; j++) {
+      newColumn.push(board[j][i]);
+    }
+
+    newColumn = shiftRowLeft(newColumn).reverse();
+
+    for (let j = 1; j < newColumn.length; j++) {
+      if (newColumn[j - 1] === newColumn[j]) {
+        newColumn[j - 1] = 0;
+        newColumn[j] = newColumn[j] * 2;
+        score += newColumn[j];
+      }
+    }
+
+    newColumn = shiftRowLeft(newColumn).reverse();
+
+    for (let j = 0; j < newColumn.length; j++) {
+      board[j][i] = newColumn[j];
+    }
+  }
+
+  if (!check) {
+    checkAndAct(check);
+  }
```

Comment:
move checking logic to a separate function. move functions should only make moves